base/src/test/java/org/aya/concrete/DesugarTest.java

```diff
@@ -54,6 +54,27 @@ public class DesugarTest {
       }""");
   }
 
+  @Test public void arrays() {
+    desugarAndPretty("""
+      open data List (A : Type)
+        | nil
+        | infixr :< A (List A)
+        
+      def infix >>= => {??}
+      
+      def simpleList => [ 1, 2, 3 ]
+      def listGenerator => [ x * y | x <- xs, y <- ys ]
+      """, """
+      data List (A : Type 0)
+        | nil
+        | :< (_ : A) (_ : List A)
+      open List hiding ()
+      def >>= => {??}
+      def simpleList => :< 1 (:< 2 (:< 3 nil))
+      def listGenerator => >>= xs (\\ (x : _) => >>= ys (\\ (y : _) => pure (x * y)))
+      """); // TODO: make desugarAndPretty does resolve [#556](https://github.com/aya-prover/aya-dev/pull/556#issuecomment-1268376917).
+  }
+
```

Comment:
I would suggest testing the desugar in some `.aya` files, like the code below that tests the BinOp desugar:

https://github.com/aya-prover/aya-dev/blob/5e317b38ea27dbe878a1f13369102ca94e6044eb/base/src/test/resources/success/src/Assoc.aya#L13-L25